scripts/build-macos.sh

```
@@ -392,3 +509,483 @@ fi
+        warn "Dylib recursion exceeded 40 passes — stopping."
+        break
+    fi
+done
+
+info "Finalizing dylib install names and cross-references..."
+for dylib in "${FRAMEWORKS_DIR}"/*.dylib; do
+    [[ -f "$dylib" ]] || continue
+    chmod u+w "$dylib"
+    install_name_tool -id "@executable_path/../Frameworks/$(basename "$dylib")" "$dylib" 2>/dev/null || true
+    while IFS= read -r libpath; do
+        local dep_base
```

Shellcheck (0.11.0)

[error] 'local' is only valid in functions.

(SC2168)